include/swift/Driver/Job.h

```diff
@@ -206,6 +206,9 @@ class CommandOutput {
   /// the sole primary input.
   StringRef getAnyOutputForType(types::ID type) const;
 
+  /// Return the whole derived output map.
+  const OutputFileMap &getDerivedOutputMap() const;
```

Comment:
The derived OFM is shared across the whole compilation; it can be written once per compilation (not once per job). There's an accessor for it on the Compilation object.

Reply:
Thanks a lot for doing this so quickly. I'll change tacks right away.

Reply:
After looking at one OFM file for all jobs, I concluded that it was worth exploring separate OFMs for two reasons: 
1. The code in the driver seems a bit cleaner because it need not add state to the Compilation to remember if the file map has been written yet (it's lazy) and what the path name was, and 
2. The cost for each batch mode frontend to read a global filemap becomes quadratic: the # of batches * the # of primaries in the whole compilation.

So, I've gone this way for now.